### PR TITLE
Display route length, and backfill the missing property as needed

### PR DESCRIPTION
**Pull request by dabreegster**
Sarah had this small request. I've stuck it below the form, but styling suggestions welcome. (All of this is going to change soon, so maybe not important.)
Screenshot from 2023-03-20 13-16-54

### CODE REVIEW

File: components/EntireScheme.svelte [hunk 1]
```diff
@@ -72,13 +73,25 @@
...
-        gjScheme.set(JSON.parse(e.target.result));
+        gjScheme.set(backfill(JSON.parse(e.target.result)));
       } catch (err) {
         window.alert(`Couldn't load scheme from a file: ${err}`);
       }
     };
     reader.readAsText(e.target.files[0]);
   }
+
+  function backfill(json) {
```
Comment (dabreegster) [inline components/EntireScheme.svelte:84]: If/when this becomes more complicated, we can split out in some way. Robin mentioned this was also an annoying difference for analysis. Before we bulk analyze schemes, we should just run the same backfill process on all the inputs.

File: tests/data/route.json [hunk 2]
```diff
@@ -0,0 +1,248 @@
+{
```
Comment (dabreegster) [inline tests/data/route.json:1]: This is just something I sketched and saved really quickly. As we get more tests, we can either have a purpose-built input for each test, or maybe more realistically, make one or two inputs that cover a larger range of test cases.

File: tests/savefiles.spec.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  // Remove the property from the test data first
+  let json = JSON.parse(await readFile("tests/data/route.json"));
+  await expect(Math.round(json.features[0].properties.length_meters)).toBe(
+    8520
+  );
+  delete json.features[0].properties.length_meters;
+  let uploadFile = JSON.stringify(json);
+
+  await page.goto("/scheme.html?authority=North Somerset");
+  await page.locator("#load_geojson").setInputFiles({
```
Comment (dabreegster) [inline tests/savefiles.spec.js:29]: See https://playwright.dev/docs/api/class-locator#locator-set-input-files for example. It's nice to pass in a local string, so we don't need to muck around with having a second copy of the input file with one tiny change

**Comment by Sparrow0hawk** [comment]
I'm testing loading previous scheme files up in this PR and running into errors. 

Files i've tried:
- File without length_meters to test backfill, [West Yorkshire Combined Authority_testy test.txt](https://github.com/acteng/atip/files/11022703/West.Yorkshire.Combined.Authority_testy.test.txt)
- FIle with length_meters, [West Yorkshire Combined Authority_magic pie.txt](https://github.com/acteng/atip/files/11022706/West.Yorkshire.Combined.Authority_magic.pie.txt)

With each of these files on loading everything appears fine but I can't browse the intervention information and adding new interventions (I can still do this) isn't displayed on the sidebar. Exporting the geojson does now include these new features but I can't annotate them in the sidebar.

In the console I get the following error:

```
HMR][Svelte] Unrecoverable HMR error in <Form>: next update will trigger a full reload [proxy.js:15:11](http://localhost:5174/node_modules/svelte-hmr/runtime/proxy.js?v=e3641821)
    logError proxy.js:15
    Proxy<Form> proxy.js:380
    create_default_slot_1 InterventionList.svelte:93
    create_slot index.mjs:69
    create_fragment5 carbon-components-svelte.js:1509
    init index.mjs:2017
    AccordionItem carbon-components-svelte.js:1782
    create_each_block InterventionList.svelte:315
    update InterventionList.svelte:426
    update_slot_base index.mjs:98
    update carbon-components-svelte.js:1143
    update carbon-components-svelte.js:1274
    update index.mjs:1193
    flush index.mjs:1160
Uncaught (in promise) TypeError: x.round is not a function
    prettyPrintMeters Form.svelte:26
    create_if_block Form.svelte:50
    create_fragment Form.svelte:49
    init index.mjs:2017
    Form Form.svelte:530
    createProxiedComponent svelte-hooks.js:341
    ProxyComponent proxy.js:242
    Proxy<Form> proxy.js:349
    create_default_slot_1 InterventionList.svelte:93
    create_slot index.mjs:69
    create_fragment5 carbon-components-svelte.js:1509
    init index.mjs:2017
    AccordionItem carbon-components-svelte.js:1782
    create_each_block InterventionList.svelte:315
    update InterventionList.svelte:426
    update_slot_base index.mjs:98
    update carbon-components-svelte.js:1143
    update carbon-components-svelte.js:1274
    update index.mjs:1193
    flush index.mjs:1160
```

File: components/Form.svelte [hunk 4]
```diff
@@ -11,13 +11,21 @@
...
   function remove() {
     gjScheme.update((gj) => {
       gj.features = gj.features.filter((f) => f.id != id);
       return gj;
     });
   }
+
+  function prettyPrintMeters(x) {
+    if (x < 1000.0) {
+      return x.round() + " m";
```
Comment (Sparrow0hawk) [inline components/Form.svelte:25]: See main thread for errors i'm getting this seem to point at this round call.

**Review by Sparrow0hawk — CHANGES_REQUESTED** [review]

**Comment by dabreegster** [comment]
Thanks so much for testing this and catching the problem! I had blind confidence in my manual and automated tests, but only hit the >1km path and didn't even think about it. I'm so used to a compiler telling me `x.round()` doesn't exist. :(   Maybe this is a strong push to switch to Typescript.

Everything looks correct to me now. I'm considering exhaustively testing <1km cases now, but it's maybe overkill. We could go back and really crank out a bunch of test cases if it becomes important.

**Pull request merged**